setup.py

```diff
@@ -1,23 +1,149 @@
+        platform = 'macosx_10_9_x86_64'
+        return python_tag, abi_tag, platform 
 
 
+def get_torch_whls(backend):
+    """Handpicks whls for torch and torchvision based on the backend.
+
+    Args:
+        backend (str): cpu or gpu
+
+    Returns:
+        list[str]: List of handpicked torch et al. whls
+    """
+    python_tag, abi_tag, platform = get_whl_info()
+    if backend == 'cpu':
+        if platform.startswith('macosx'):
+            _torch_whl = f'-{python_tag}-none-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{_torch_whl}'
+            _torchvision_whl = f'-{python_tag}-{abi_tag}-{platform}.whl'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/torchvision-0.9.0{_torchvision_whl}'
+            return torch_whl, torchvision_whl
+        else:
+            whl = f'%2B{backend}-{python_tag}-{abi_tag}-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{whl}'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/cpu/torchvision-0.9.0{whl}'
+            return torch_whl, torchvision_whl
+    elif backend == 'gpu':
+        if platform.startswith('macosx'):
+            # MacOS Binaries don't support CUDA, install from source if CUDA is needed
+            # install on CPU by default
+            _torch_whl = f'-{python_tag}-none-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{_torch_whl}'
+            _torchvision_whl = f'-{python_tag}-{abi_tag}-{platform}.whl'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/torchvision-0.9.0{_torchvision_whl}'
+            return torch_whl, torchvision_whl
+        else:
+            backend = 'cu111'
+            whl = f'%2B{backend}-{python_tag}-{abi_tag}-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cu111/torch-1.8.1{whl}'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/cu111/torchvision-0.9.1{whl}'
+            return torch_whl, torchvision_whl
+
+
+extra_requirements = {
+    'cpu': [
+        requirements,
+        'onnxruntime==1.7.0',
+        get_torch_whls(backend='cpu')
+    ],
+    'gpu': [
+        requirements,
+        get_torch_whls(backend='gpu'),
+    ],
+    'docs': [
+        # pin sphinx to match what RTD uses:
+        # https://github.com/readthedocs/readthedocs.org/blob/ecac31de54bbb2c100f933e86eb22b0f4389ba84/requirements/pip.txt#L16
+        'sphinx',
+        'sphinx_rtd_theme',
+        'sphinx-tabs==3.2.0',
+        'sphinx-toolbox==2.15.2',
+        'sphinx-jsonschema~=1.16',
+        'pypandoc',
+    ],
+    'tests': [
+        'pytest~=6.2',
+        'pytest-cov',
+        'pytest-ordering~=0.6',
+        'pytest-console-scripts~=1.1',
+        'coverage',
+        'coveralls',
+    ],
+    'contrib': [
+        'pre-commit>=2.10.1',
+        'flake8',    
+    ]
```

Comment:
~~These are pretty much required for devs as well? So probably in the PR on documentation update, we should mention this on the installation for dev section.~~

I saw the lines below. Perfect!